### ISSUE TITLE
ASL fails to compile

### ISSUE THREAD

**Issue by jalving** [Argonne-National-Laboratory/PIPS#45]
I've been running into an issue compiling the latest ASL and I'm not sure where to report this.
I attached the output I'm getting.  This is with a previously working PIPS installation.  It seems the wget is grabbing a new ASL that introduced a compile error.  

Can anyone reproduce this error?

-------------------------------

> jordan@spartacus-III:/opt/PIPS2/ThirdPartyLibs/ASL$ sudo ./wgetASL.sh 
>  
> ### Downloading the third party packages for PIPS-NLP:
>  
> ### Downloading Ampl Solver Library (ASL):
> --2019-01-09 14:11:14--  http://www.ampl.com/netlib/ampl/solvers.tgz
> Resolving www.ampl.com (www.ampl.com)... 70.38.71.71
> Connecting to www.ampl.com (www.ampl.com)|70.38.71.71|:80... connected.
> HTTP request sent, awaiting response... 301 Moved Permanently
> Location: https://ampl.com/netlib/ampl/solvers.tgz [following]
> --2019-01-09 14:11:15--  https://ampl.com/netlib/ampl/solvers.tgz
> Resolving ampl.com (ampl.com)... 70.38.71.71
> Connecting to ampl.com (ampl.com)|70.38.71.71|:443... connected.
> HTTP request sent, awaiting response... 200 OK
> Length: 349689 (341K) [application/x-gzip]
> Saving to: ‘solvers.tar.gz’
> 
> solvers.tar.gz                                       100%[====================================================================================================================>] 341.49K   931KB/s    in 0.4s    
> 
> 2019-01-09 14:11:15 (931 KB/s) - ‘solvers.tar.gz’ saved [349689/349689]
> 
> ### ASL: Download Successful.
> 
> ln: failed to create symbolic link './src/solvers': File exists
> Applying patch for #define filename in asl.h, which is incompatible with mpi.h.
> rm -f amplsolver.a *.o arith.h details.c float.h stdio1.h
> cc -c -O -DASL_NO_FPINITMT fpinit.c
> cat stdio1.h0 >stdio1.h
> cc -O arithchk.c fpinit.o -lm ||\
>  cc -DNO_LONG_LONG -O arithchk.c fpinit.o -lm ||\
>  cc -DNO_SSIZE_T -O arithchk.c fpinit.o -lm ||\
>  cc -DNO_LONG_LONG -DNO_SSIZE_T -O arithchk.c fpinit.o -lm
> ./a.out >arith.h
> rm -f a.out arithchk.o fpinit.o
> sed "s!System_details!`uname -sm`!" details.c0 >details.c
> cc -c -O funcadd0.c
> cc -c -O asldate.c atof.c auxinfo.c avltree.c b_search.c basename.c bscanf.c com2eval.c comeval.c con1ival.c con2ival.c con2val.c conadj.c conpval.c conscale.c conval.c degree.c derprop.c details.c dtoa1.c duthes.c dynlink.c f_read.c fg_read.c fg_write.c fgh_read.c fpecatch.c fpinit.c fullhes.c func_add.c funcadd1.c g_fmt.c genrowno.c getenv.c getstub.c htcl.c indic_cons.c jac0dim.c jac2dim.c jacdim.c jacinc.c jacinc1.c libnamsave.c mach.c mainexit.c mip_pri.c misc.c mpec_adj.c mqpcheckv.c mypow.c names.c nl_obj.c nqpcheck.c nqpcheckZ.c obj2val.c obj_adj.c obj_prec.c objconst.c objval.c objval_.c op_type.c pfg_read.c pfghread.c printf.c pshvprod.c punknown.c qp_read.c qpcheck.c qpcheckZ.c qsortv.c readsol.c repwhere.c rops.c rops2.c sigcatch.c sos_add.c sphes.c sscanf.c stderr.c studchk0.c suf_sos.c value.c writesol.c wrtsol_.c ws_desc.c wsu_desc.c x2check.c xectim.c xp1known.c xp2known.c
> jac0dim.c: In function ‘jac0dim_ASL’:
> jac0dim.c:207:10: error: ‘Edaginfo {aka struct Edaginfo}’ has no member named ‘nlc0’
>     asl->i.nlc0 = nlc;
>           ^
> jac0dim.c:208:10: error: ‘Edaginfo {aka struct Edaginfo}’ has no member named ‘nlo0’
>     asl->i.nlo0 = nlo;
>           ^
> sphes.c: In function ‘saveog’:
> sphes.c:521:22: error: ‘Edaginfo {aka struct Edaginfo}’ has no member named ‘nlc0’
>    for(pe = p + asl->i.nlc0; p < pe; p++)
>                       ^
> sphes.c:540:22: error: ‘Edaginfo {aka struct Edaginfo}’ has no member named ‘nlc0’
>    for(pe = p + asl->i.nlc0; p < pe; p++)
>                       ^
> sphes.c: In function ‘restog’:
> sphes.c:567:22: error: ‘Edaginfo {aka struct Edaginfo}’ has no member named ‘nlc0’
>    for(pe = p + asl->i.nlc0; p < pe; p++)
>                       ^
> sphes.c: In function ‘sphes_setup_ASL’:
> sphes.c:764:15: error: ‘Edaginfo {aka struct Edaginfo}’ has no member named ‘nlc0’
>   nlc0 = asl->i.nlc0;
>                ^
> sphes.c: In function ‘sphes_ASL’:
> sphes.c:1091:15: error: ‘Edaginfo {aka struct Edaginfo}’ has no member named ‘nlc0’
>   nlc0 = asl->i.nlc0;
>                ^
> sphes.c:1154:22: error: ‘Edaginfo {aka struct Edaginfo}’ has no member named ‘nlc0’
>    for(pe = p + asl->i.nlc0; p < pe; p++, y1++)
>                       ^
> makefile:185: recipe for target 'amplsolver.a' failed
> make: *** [amplsolver.a] Error 1

------------------------------

**Comment by cnpetra** [comment]
that's because of some recent changes in the ASL. I decided to not support AMPL/ASL because I am not aware of anyone using it. I will be removing from the build...

**Comment by jalving** [comment]
It looks like the Ipopt build is pulling an old ASL v1.3.0. I could change that script if you want to keep supporting it.  It's just 2 lines.
```
coinasl=1.3.0
if wget -O solvers.tar.gz https://github.com/ampl/mp/archive/$coinasl.tar.gz
```

**Comment by nychiang** [comment]
I think it is better to keep it if this just needs 2 lines modification.
In addition, the Argonne folks would like to use it for the debug reason.
I will come back to this in the beginning of Feb, and also add support to the users' dual initial values.

**Comment by cnpetra** [comment]
@nychiang : someone else reported the bug (on a private channel). How's the fix going?

**Comment by jalving** [comment]
@cnpetra @nychiang 
Since it's an easy change, I can make the pull request.  Should only take a few minutes.

**Comment by nychiang** [comment]
Thanks Jordan. I will test it, on Thursday or Friday. Cheers

**Comment by cnpetra** [comment]
addressed by PR #48; it is merged in the master now, together with a fix for the mumps/parmetis build  PR #46

**Comment by nychiang** [comment]
I did a clean build and it is totally fine.
Cheers,

NY
